skactiveml/pool/tests/test_qbc.py

```diff
@@ -1,104 +1,215 @@
-    com_probas = np.zeros((len(estimators), len(X_cand), ensemble.n_classes_))
-    for i, e in enumerate(estimators):
-        com_probas[i, :, :] = e.predict_proba(X_cand)
-
-    consensus = np.sum(com_probas, axis=0) / len(estimators)
-    scores = np.zeros((len(X_cand)))
-    for i, x in enumerate(X_cand):
-        for c in range(len(estimators)):
-            for y in range(ensemble.n_classes_):
-                with np.errstate(divide='ignore', invalid='ignore'):
-                    if com_probas[c, i, y] != 0.0:
-                        scores[i] += com_probas[c, i, y] * np.log(com_probas[c, i, y] / consensus[i, y])
-    scores = scores / ensemble.n_classes_
-    return scores
-
-
-def vote_entropy(ensemble, X_cand, classes):
-    estimators = ensemble.estimators_
-    votes = np.zeros((len(X_cand), len(estimators)))
-    for i, model in enumerate(estimators):
-        votes[:, i] = model.predict(X_cand)
-
-    vote_count = np.zeros((len(X_cand), len(classes)))
-    for i in range(len(X_cand)):
-        for c in range(len(classes)):
-            for m in range(len(estimators)):
-                vote_count[i, c] += (votes[i, m] == c)
-
-    vote_entropy = np.zeros(len(X_cand))
-    for i in range(len(X_cand)):
-        for c in range(len(classes)):
-            if vote_count[i, c] != 0:
-                a = vote_count[i, c] / len(estimators)
-                vote_entropy[i] += a * np.log(a)
-    vote_entropy *= -1 / np.log(len(estimators))
-
-    return vote_entropy
+        selector = QBC(clf=self.clf, random_state=self.random_state)
+        best_indices1, utilities1 = selector.query(**self.kwargs,
+                                                   return_utilities=True)
+        selector = QBC(clf=self.clf, random_state=self.random_state)
+        best_indices2, utilities2 = selector.query(**self.kwargs,
+                                                   return_utilities=True)
+        np.testing.assert_array_equal(utilities1, utilities2)
+        np.testing.assert_array_equal(best_indices1, best_indices2)
+
+
+class TestAverageKlDivergence(unittest.TestCase):
+    def setUp(self):
+        self.probas = np.array([[[0.3, 0.7], [0.4, 0.6]],
+                                [[0.2, 0.8], [0.5, 0.5]]])
+        self.scores = np.array([0.00670178182226764, 0.005059389928987596])
+
+    def test_param_probas(self):
+        self.assertRaises(ValueError, average_kl_divergence, 'string')
+        self.assertRaises(ValueError, average_kl_divergence, 1)
+        self.assertRaises(ValueError, average_kl_divergence,
+                          np.ones((1,)))
+        self.assertRaises(ValueError, average_kl_divergence,
+                          np.ones((1, 1)))
+        self.assertRaises(ValueError, average_kl_divergence,
+                          np.ones((1, 1, 1, 1)))
+
+    def test_average_kl_divergence(self):
+        scores = average_kl_divergence(self.probas)
+        np.testing.assert_array_equal(scores, self.scores)
```

Comment:
it would be good to test against special cases. eg zero probabilities, equal probabilities, ...